src/Psalm/Internal/PhpVisitor/ReflectorVisitor.php

```diff
@@ -2133,6 +2134,12 @@ private function registerFunctionLike(PhpParser\Node\FunctionLike $stmt, $fake_m
 
         $doc_comment = $stmt->getDocComment();
 
+
+        if ($class_storage && ! $class_storage->is_trait) {
```

Comment:
We don't take the internal nature of the trait into account, because that would fail these existing test cases for valid code:

https://github.com/vimeo/psalm/blob/b27560617c88bed9df0ee6dc80ab9487313a4b95/tests/PsalmInternalAnnotationTest.php#L221 https://github.com/vimeo/psalm/blob/b27560617c88bed9df0ee6dc80ab9487313a4b95/tests/InternalAnnotationTest.php#L175